### DISCUSSION TITLE
Doubt in the aggregation step of Heterogenous networks

### DISCUSSION THREAD

**Discussion by Vaibhav3006** [pyg-team/pytorch_geometric#3929] [Q&A]
Hi, I am relatively new to heterogeneous information networks and I was going through some of the papers around GraphSage and MetaPath2vec. My question has 2 parts

1.  While aggregating neighborhood structure information ( matrix we get after MetaPath based deep walks ) with the node features matrix, how do we address the situation when the shape of the feature matrix is different from the matrix we get after deep walks ?
2. When performing such aggregations across a heterogeneous graph, how do we solve for cases when the nodes of 2 different type ( say Author and paper ) have different-sized feature vectors. Do we only use the feature vectors of similar type nodes while aggregating neighborhood matrix information?

Thanks in advance.

**Comment by rusty1s** [comment]
Sorry for the late reply. I hope I understand your questions correctly.

In general, in order to apply aggregations across different types, the shape of feature matrices need to match. As such, at some point in time, these have to be transformed to the same dimensionality. In heterogeneous GNNs, this is typically achieved by utilizing the same hidden feature dimensionality across all node types. Nonetheless, the input channels across node types can be different at the start, and individual transformations are used to map them to the same size.
